### PR TITLE
chat notifications: Prevent localPlayer highlights and notifications

### LINKED ISSUE #13238: Self/own messages triggering chat notifications
**Describe the bug**
When you type a message containing phrases you have highlighted to send notifications
>Chat notifications on
>Highlight words
>Notify on highlight
>Allow desktop notifications (not necessary but annoying)

you will receive a desktop notification every time your chat (including autochat!) says the phrase INCLUDING if your username has the phrase/word in it for example you want to know if someone says "whip" if your name is "whip" and/or u say "whip 5m" (including on autochat - more likely to happen in this scenario) u will receive notifications you definitely do not want.

**To Reproduce**
I believe this is example above sorry

**Expected behavior**
No notifications for own messages, alternatively a button to allow self-messages to be excluded from notifications.

**Environment (please complete the following information):**
 - OS: windows 10 home unactivated
 - RuneLite version: 17.0
 - Launcher version: 2.1.6

### PR DESCRIPTION
**Pull request by Broooklyn**
Fixes #13238

### CODE REVIEW

**Review by Nightfirecat — CHANGES_REQUESTED** [review]
The tests for this plugin currently pass regardless of whether this is applied, so some test changes are needed to provide coverage. You should be able to mock a localPlayer response in whatever added tests you provide without impacting the coverage of the existing tests.

File: runelite-client/src/main/java/net/runelite/client/plugins/chatnotifications/ChatNotificationsPlugin.java [hunk 1]
```diff
@@ -172,6 +172,17 @@ public void onChatMessage(ChatMessage chatMessage)
...
 				}
 				break;
+			case PRIVATECHATOUT:
+				return;
+			case PUBLICCHAT:
+			case FRIENDSCHAT:
+			case AUTOTYPER:
+				if (client.getLocalPlayer() != null && chatMessage.getName().equals(client.getLocalPlayer().getName()))
+				{
+					messageNode.setValue(messageNode.getValue());
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/chatnotifications/ChatNotificationsPlugin.java:182]: Behavior-wise, this is a no-op?

File: runelite-client/src/test/java/net/runelite/client/plugins/chatnotifications/ChatNotificationsPluginTest.java [hunk 2]
```diff
@@ -286,4 +289,48 @@ public void testHighlightOwnNameNbsp()
...
+		final String localPlayerName = "Brooklyn";
+
+		MessageNode messageNode = mock(MessageNode.class);
+
+		Player localPlayer = mock(Player.class);
+		when(client.getLocalPlayer()).thenReturn(localPlayer);
+		when(localPlayer.getName()).thenReturn(localPlayerName);
+
+		lenient().when(config.highlightOwnName()).thenReturn(true);
+		lenient().when(config.notifyOnHighlight()).thenReturn(true);
```
Comment (Nightfirecat) [inline runelite-client/src/test/java/net/runelite/client/plugins/chatnotifications/ChatNotificationsPluginTest.java:305]: Neither test is specifically testing notifications; only the name/words highlight configs need to be mocked in the added tests.

File: runelite-client/src/main/java/net/runelite/client/plugins/chatnotifications/ChatNotificationsPlugin.java [hunk 3]
```diff
@@ -172,6 +172,16 @@ public void onChatMessage(ChatMessage chatMessage)
 					notifier.notify(Text.removeTags(chatMessage.getName()) + ": " + chatMessage.getMessage());
 				}
 				break;
+			case PRIVATECHATOUT:
+				return;
+			case PUBLICCHAT:
+			case FRIENDSCHAT:
+			case AUTOTYPER:
+				if (client.getLocalPlayer() != null && chatMessage.getName().equals(client.getLocalPlayer().getName()))
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/chatnotifications/ChatNotificationsPlugin.java:180]: This fails for players with spaces in their name. `Player#getName()` appears to return a normal space, while `ChatMessage#getName()` uses a nbsp. It's likely safest to assume both strings must be run through `Text#toJagexName()`.

The test case covering this should be updated as well.

**Review by Nightfirecat — CHANGES_REQUESTED** [review]

**Comment by Broooklyn** [comment]
Updated per your review.  It was also necessary to wrap `chatMessage.getName()` in `Text.removeTags()` in addition to `Text.toJagexName()` to account for Ironmen players.

That said, I am fairly certain `Text#toJagexName()` does not have any effect on `client.getLocalPlayer().getName()` so I have ommited it from that side of the comparison.

File: runelite-client/src/main/java/net/runelite/client/plugins/chatnotifications/ChatNotificationsPlugin.java [hunk 4]
```diff
@@ -172,6 +172,18 @@ public void onChatMessage(ChatMessage chatMessage)
...
 				}
 				break;
+			case PRIVATECHATOUT:
+				return;
+			case MODCHAT:
+			case PUBLICCHAT:
+			case FRIENDSCHAT:
+			case AUTOTYPER:
+			case MODAUTOTYPER:
+				if (client.getLocalPlayer() != null && Text.toJagexName(Text.removeTags(chatMessage.getName())).equals((client.getLocalPlayer().getName())))
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/chatnotifications/ChatNotificationsPlugin.java:182]: One too many parens around the localplayer name call
```suggestion
				if (client.getLocalPlayer() != null && Text.toJagexName(Text.removeTags(chatMessage.getName())).equals(client.getLocalPlayer().getName()))
```

**Pull request merged**